charmcraft/parts.py

```diff
@@ -257,6 +266,16 @@ def post_build_callback(self, step_info):
         """Collect metrics left by charm_builder.py."""
         instrum.merge_from(env.get_charm_builder_metrics_path())
 
+    def _get_os_special_priority_paths(self) -> list:
```

Comment:
I really think we should just return a `Optional[str]` instead of a list here
if you're really set on returning a list, this annotation should be `List[str]`

Reply:
I can go for str, don't seems use a lot atm.

Reply:
alright!